hazelcast/src/main/java/com/hazelcast/query/impl/getters/AbstractMultiValueGetter.java

```diff
@@ -156,15 +156,23 @@ private Object getParentObject(Object obj) throws Exception {
 
     private void reduceArrayInto(MultiResult collector, Object[] currentObject) {
         Object[] array = currentObject;
-        for (int i = 0; i < array.length; i++) {
-            collector.add(array[i]);
+        if (array.length == 0) {
```

Comment:
can you describe a rationale behind these changes? (this and the method with collections)